c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -266,25 +276,32 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
                         self.visit_slice_as_ptr(elem_ty, &args[0], pl_ty);
                         return;
                     }
-                    _ => {}
-                }
 
-                // General case: cast `args` to match the signature of `func`.
-                let poly_sig = func_ty.fn_sig(tcx);
-                let sig = tcx.erase_late_bound_regions(poly_sig);
-
-                for (i, _op) in args.iter().enumerate() {
-                    if i >= sig.inputs().len() {
-                        // This is a call to a variadic function, and we've gone past the end of
-                        // the declared arguments.
-                        // TODO: insert a cast to turn `op` back into its original declared type
-                        // (i.e. upcast the chosen reference type back to a raw pointer)
-                        continue;
+                    Callee::LocalDef { def_id, substs: _ } => {
+                        // TODO: handle substs (if nonempty)
+                        if let Some(lsig) = self.acx.gacx.fn_sigs.get(&def_id) {
```

Comment:
There shouldn't be any cases where this returns `None`, right?  Those should be `UnknownDef`s.  In other cases like this, we do `.fn_sigs.get(&def_id).unwrap()` or similar.

Reply:
Looks to be fixed in 6d099d07176a24079c9b3ba8d51d1d29b4979bb2.